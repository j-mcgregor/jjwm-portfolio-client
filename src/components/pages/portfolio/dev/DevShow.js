/* eslint-disable operator-linebreak */
import React from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import projectImages from '../../../../images';

function DevShow({ projects }) {
  const { devId } = useParams();
  const project = projects.find(p => p.id === devId) || {};

  return (
    <div className="row">
      <div className="col-md-8">
        <h3 className="py-4">{project.name}</h3>
      </div>
      <div className="col-md-8">
        <img
          src={projectImages[devId - 1]}
          alt={project.name}
          className="screenshot"
        />
      </div>
      <div className="col-8 py-4">
        {project.url && (
          <a href={project.url} target="_blank" rel="noopener noreferrer">
            <i className="fas fa-globe fa-2x mr-3" />
          </a>
        )}
        {project.twitter && (
          <a href={project.twitter} target="_blank" rel="noopener noreferrer">
            <i className="fab fa-twitter fa-2x mr-3" />
          </a>
        )}
        {project.facebook && (
          <a href={project.facebook} target="_blank" rel="noopener noreferrer">
            <i className="fab fa-facebook fa-2x mr-3" />
          </a>
        )}
        {project.github && (
          <a href={project.github} target="_blank" rel="noopener noreferrer">
            <i className="fab fa-github fa-2x mr-3" />
          </a>
        )}
      </div>
      <div className="col-8">
        <p>{project.description}</p>
      </div>
    </div>
  );
}

DevShow.propTypes = {
  projects: PropTypes.arrayOf(PropTypes.shape({}))
};

DevShow.defaultProps = {
  projects: []
};

export default DevShow;
