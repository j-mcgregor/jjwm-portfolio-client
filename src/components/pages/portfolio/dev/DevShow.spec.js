import React from 'react';
import { shallow } from 'enzyme';
import DevShow from './DevShow';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({ devId: '1' })
}));

describe('DevShow', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render', () => {
    const props = {
      projects: [
        {
          id: '1',
          name: 'Test 1',
          url: 'www.test.com',
          twitter: 'www.twitter.com/twitter1',
          facebook: 'www.facebook.com/facebook1',
          github: 'www.github.com/github1',
          description: 'lorem'
        },
        {
          id: '2',
          name: 'Test 2',
          url: 'www.test.com',
          twitter: 'www.twitter.com/twitter1',
          facebook: 'www.facebook.com/facebook1',
          github: 'www.github.com/github1',
          description: 'lorem'
        },
        {
          id: '3',
          name: 'Test 3',
          url: 'www.test.com',
          twitter: 'www.twitter.com/twitter1',
          facebook: 'www.facebook.com/facebook1',
          github: 'www.github.com/github1',
          description: 'lorem'
        }
      ]
    };
    const wrapper = shallow(<DevShow {...props} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('a').get(0).props.href).toBe('www.test.com');
    expect(wrapper.find('a').get(1).props.href).toBe(
      'www.twitter.com/twitter1'
    );
    expect(wrapper.find('a').get(2).props.href).toBe(
      'www.facebook.com/facebook1'
    );
    expect(wrapper.find('a').get(3).props.href).toBe('www.github.com/github1');
  });
});
