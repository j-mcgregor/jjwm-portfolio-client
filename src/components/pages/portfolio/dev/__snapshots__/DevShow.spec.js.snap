// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevShow should render 1`] = `ShallowWrapper {}`;
